### PR TITLE
Roll engine to 50c2e69da with goldens fix

### PR DESCRIPTION
**Pull request by liyuqian**
Goldens updated due to Skia AA changes. We previously also made a mistake by having a golden test run on MacOS.

git log 3ffa36295234..50c2e69da --no-merges --oneline
50c2e69da Roll src/third_party/skia 1b62fad4b188..b27a9cf2f4a8 (1 commits) (flutter/engine#6514)
8757bf2a1 Roll src/third_party/skia 05caa69a3f5a..1b62fad4b188 (2 commits) (flutter/engine#6513)
c2fd10ad1 Roll src/third_party/skia 2a138544646d..05caa69a3f5a (1 commits) (flutter/engine#6512)

### CODE REVIEW

File: packages/flutter/test/cupertino/segmented_control_test.dart [hunk 1]
```diff
@@ -1314,7 +1314,7 @@ void main() {
       find.byType(RepaintBoundary),
       matchesGoldenFile('segmented_control_test.0.0.png'),
     );
-  }, skip: !Platform.isMacOS);
+  }, skip: !Platform.isLinux);
```
Comment (GaryQian) [inline packages/flutter/test/cupertino/segmented_control_test.dart]: There should be another place where this needs to be changed

Comment (GaryQian) [reply packages/flutter/test/cupertino/segmented_control_test.dart]: Line 1276

Comment (liyuqian) [reply packages/flutter/test/cupertino/segmented_control_test.dart]: Good catch! Will update shortly.

**Review by GaryQian — APPROVED** [review]

**Review by gspencergoog — APPROVED** [review]

**Comment by aam** [comment]
cc @rmacnak-google who is finishing/starting the dart roll

**Pull request merged**